Failed to upload files to the mount point of the ufs cluster configured by parameters

**Alluxio Version:**
2.7.1

**Describe the bug**
Mount a ufs path using option `alluxio.underfs.hdfs.configuration`. The mount was successful, but the file upload failed.


**To Reproduce**

1. Mount a ufs path using option `alluxio.underfs.hdfs.configuration`.
2. There is no configuration file on the worker node.
3. Use `copyFromLocal` upload a file and it will be failed.
4. Add configuration files for worker and re-upload the file, it still fails.

**Expected behavior**
After adding the configuration files to the worker, the file can be re-uploaded without restarting the worker node.


**Are you planning to fix it**
yes. 

**Additional context**
Error log:
```
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalArgumentException: java.net.UnknownHostException: XX
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3951)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3974)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4958)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem.getFs(HdfsUnderFileSystem.java:831)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem.createDirect(HdfsUnderFileSystem.java:287)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem.create(HdfsUnderFileSystem.java:279)
        at alluxio.underfs.ConsistentUnderFileSystem.createNonexistingFile(ConsistentUnderFileSystem.java:46)
        at alluxio.underfs.UnderFileSystemWithLogging$8.call(UnderFileSystemWithLogging.java:224)
        at alluxio.underfs.UnderFileSystemWithLogging$8.call(UnderFileSystemWithLogging.java:221)
        at alluxio.underfs.UnderFileSystemWithLogging.call(UnderFileSystemWithLogging.java:1213)
        at alluxio.underfs.UnderFileSystemWithLogging.createNonexistingFile(UnderFileSystemWithLogging.java:221)
        at alluxio.worker.grpc.UfsFileWriteHandler.createUfsFile(UfsFileWriteHandler.java:166)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:139)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:49)
        at alluxio.worker.grpc.AbstractWriteHandler.writeData(AbstractWriteHandler.java:281)
        at alluxio.worker.grpc.AbstractWriteHandler.lambda$writeDataMessage$1(AbstractWriteHandler.java:179)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at alluxio.worker.grpc.GrpcExecutors$ImpersonateThreadPoolExecutor.lambda$execute$0(GrpcExecutors.java:159)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: java.net.UnknownHostException: XX
        at org.apache.hadoop.security.SecurityUtil.buildTokenService(SecurityUtil.java:444)
        at org.apache.hadoop.hdfs.NameNodeProxiesClient.createProxyWithClientProtocol(NameNodeProxiesClient.java:142)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:375)
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:309)
        at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:168)
        at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3269)
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:472)
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:356)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem$1.load(HdfsUnderFileSystem.java:186)
        at alluxio.underfs.hdfs.HdfsUnderFileSystem$1.load(HdfsUnderFileSystem.java:172)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045)
        ... 21 more
Caused by: java.net.UnknownHostException: XX
        ... 35 more
```

--- a/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
+++ b/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
@@ -160,6 +160,7 @@ public abstract class AbstractUfsManager implements UfsManager {
       if (useManagedBlocking) {
         fs = new ManagedBlockingUfsForwarder(fs);
       }
+      mCloser.register(fs);
       try {
         connectUfs(fs);
         tryUseFileSystem(fs, ufsUri.getPath());
@@ -174,7 +175,6 @@ public abstract class AbstractUfsManager implements UfsManager {
         // This shouldn't occur unless our synchronization is incorrect
         LOG.warn("UFS already existed in UFS manager");
       }
-      mCloser.register(fs);
       return fs;
     }
   }

--- a/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
+++ b/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
@@ -168,7 +168,7 @@ public abstract class AbstractUfsManager implements UfsManager {
       mCloser.register(fs);
       try {
         connectUfs(fs);
-        ufsUriExists(fs, ufsUri.getPath());
+        tryUseFileSystem(fs, ufsUri.getPath());
       } catch (Exception e) {
         mUnderFileSystemMap.remove(key);
         String message = String.format(
@@ -189,15 +189,14 @@ public abstract class AbstractUfsManager implements UfsManager {
   protected abstract void connectUfs(UnderFileSystem fs) throws IOException;
 
   /**
-   * Used to check whether the filesystem is available.
+   * To check whether the filesystem is available by calling exists.
    *
    * @param fs the filesystem
    * @param uriPath the UFS path
-   * @return true if the file exits
    * @throws Exception
    */
-  private boolean ufsUriExists(UnderFileSystem fs, String uriPath) throws Exception {
-    return fs.exists(uriPath);
+  private void tryUseFileSystem(UnderFileSystem fs, String uriPath) throws Exception {
+    fs.exists(uriPath);
   }
 
   @Override

--- a/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
+++ b/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
@@ -160,23 +160,21 @@ public abstract class AbstractUfsManager implements UfsManager {
       if (useManagedBlocking) {
         fs = new ManagedBlockingUfsForwarder(fs);
       }
-
-      if (mUnderFileSystemMap.putIfAbsent(key, fs) != null) {
-        // This shouldn't occur unless our synchronization is incorrect
-        LOG.warn("UFS already existed in UFS manager");
-      }
-      mCloser.register(fs);
       try {
         connectUfs(fs);
         tryUseFileSystem(fs, ufsUri.getPath());
       } catch (Exception e) {
-        mUnderFileSystemMap.remove(key);
         String message = String.format(
             "Failed to perform initial connect to UFS %s: %s", ufsUri, e);
         recorder.record(message);
         LOG.warn(message);
         throw new RuntimeException(e);
       }
+      if (mUnderFileSystemMap.putIfAbsent(key, fs) != null) {
+        // This shouldn't occur unless our synchronization is incorrect
+        LOG.warn("UFS already existed in UFS manager");
+      }
+      mCloser.register(fs);
       return fs;
     }
   }

--- a/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
+++ b/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
@@ -192,11 +192,11 @@ public abstract class AbstractUfsManager implements UfsManager {
    * To check whether the filesystem is available by calling exists.
    *
    * @param fs the filesystem
-   * @param uriPath the UFS path
+   * @param ufsPath the UFS path
    * @throws Exception
    */
-  private void tryUseFileSystem(UnderFileSystem fs, String uriPath) throws Exception {
-    fs.exists(uriPath);
+  private void tryUseFileSystem(UnderFileSystem fs, String ufsPath) throws Exception {
+    fs.exists(ufsPath);
   }
 
   @Override

--- a/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
+++ b/core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java
@@ -168,11 +168,14 @@ public abstract class AbstractUfsManager implements UfsManager {
       mCloser.register(fs);
       try {
         connectUfs(fs);
-      } catch (IOException e) {
+        ufsUriExists(fs, ufsUri.getPath());
+      } catch (Exception e) {
+        mUnderFileSystemMap.remove(key);
         String message = String.format(
             "Failed to perform initial connect to UFS %s: %s", ufsUri, e);
         recorder.record(message);
         LOG.warn(message);
+        throw new RuntimeException(e);
       }
       return fs;
     }
@@ -184,6 +187,18 @@ public abstract class AbstractUfsManager implements UfsManager {
    *  {@link UnderFileSystem#connectFromWorker(String)} depending on the running process.
    */
   protected abstract void connectUfs(UnderFileSystem fs) throws IOException;
+
+  /**
+   * Used to check whether the filesystem is available.
+   *
+   * @param fs the filesystem
+   * @param uriPath the UFS path
+   * @return true if the file exits
+   * @throws Exception
+   */
+  private boolean ufsUriExists(UnderFileSystem fs, String uriPath) throws Exception {
+    return fs.exists(uriPath);
+  }
 
   @Override
   public void addMount(long mountId, final AlluxioURI ufsUri,

